app/Http/Controllers/EventController.php

```diff
@@ -83,6 +83,14 @@ public function destroyAuthParticipant($event_id)
         $participant = Participant::where('email', Auth::user()->email)->first();
         $participant->joinEvents()->where('event_id', $event_id)->detach();
         return redirect()->route('events.show', $event_id);
+
+    }
+  
+    public function cancel(Event $event)
+    {
+        $participant = Participant::where('email', Auth::user()->email)->first();
+        $participant->joinEvents()->detach($event);
```

Comment:
$this should be `detach($event->id)`

Reply:
and also please remove the entire
```
 public function __construct(Level $level, Category $category, Meeting $meeting, Room $room){
        $this->level = $level;
        $this->category = $category;
        $this->meeting = $meeting;
        $this->room = $room;
    }
```

if you use something like `$this->event->all()` you can simply replace that with `Event::all()` and its shorter

Reply:
I fix  detach($event) to detach($event->id).

 "public function __construct 〜〜" is  in MeetingController.
I delete this parts and replace with like Event::all().